apiserver/pkg/http/client_test.go

```diff
@@ -1,17 +1,65 @@
 package http
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
 	"net/http"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 	rpcStatus "google.golang.org/genproto/googleapis/rpc/status"
 
+	"github.com/ray-project/kuberay/apiserver/pkg/util"
 	api "github.com/ray-project/kuberay/proto/go_client"
 )
 
+type mockTransport struct {
+	statusErr      *rpcStatus.Status
+	body           string
+	statusSequence []int
+	callCount      int
+	returnDoError  bool
+}
+
+func (m *mockTransport) RoundTrip(_ *http.Request) (*http.Response, error) {
+	if m.returnDoError {
+		return nil, errors.New("mock go error")
+	}
+
+	index := m.callCount
+	status := http.StatusOK
+	if index < len(m.statusSequence) {
+		status = m.statusSequence[index]
+	}
+
+	respBody := m.body
+	if status != http.StatusOK && m.statusErr != nil {
+		statusMsg := fmt.Sprintf(`{"code": %d, "message": "%s"}`, m.statusErr.Code, m.statusErr.Message)
+		respBody = statusMsg
+	}
+
+	m.callCount++
```

Comment:
nit: do you mind put increment right after L33? I'm worried we might early return in the middle in the future.